### PR TITLE
[BE-11] add products to recipe

### PR DESCRIPTION
**Pull request by plokijro**
https://trello.com/c/OTt1roWz/36-be-11-user-can-add-products-to-recipes-l

### CODE REVIEW

File: db/migrate/20210822130020_create_ingredients.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateIngredients < ActiveRecord::Migration[6.1]
+  def change
+    create_table :ingredients do |t|
+      t.integer :quantity
```
Comment (h4b00) [inline db/migrate/20210822130020_create_ingredients.rb:4]: `null: false` ? 
or `default: 0`

File: spec/controllers/api/v1/recipes_controller_spec.rb [hunk 2]
```diff
@@ -27,6 +27,10 @@
     it 'recipe has comments' do
       expect(JSON.parse(response.body)).to include('comments' => recipe.comments.to_a)
     end
+
+    it 'recipe has products' do
+      expect(JSON.parse(response.body)).to include('products' => recipe.comments)
+    end
```
Comment (h4b00) [inline spec/controllers/api/v1/recipes_controller_spec.rb:33]: ?

File: spec/models/recipe_spec.rb [hunk 3]
```diff
@@ -9,6 +9,8 @@
 
   describe 'relationships' do
     it { is_expected.to have_many(:comments) }
+    it { is_expected.to have_many(:ingredients) }
```
Comment (h4b00) [inline spec/models/recipe_spec.rb:12]: missing `.dependent`
https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_record/association_matcher.rb#L149

File: app/serializers/recipe_serializer.rb [hunk 4]
```diff
@@ -39,4 +38,17 @@ def price
...
+      product = {
+        id: ingredient.product_id,
+        name: object.products.find(ingredient.product_id).name,
+        quantity: ingredient.quantity
+      }
+      products_array.push(product)
+    end
+
+    products_array
+  end
```
Comment (h4b00) [inline app/serializers/recipe_serializer.rb:53]: You can write it all just by using a map.
without assigning an empty array and pushing to it.

Comment (h4b00) [reply app/serializers/recipe_serializer.rb:53]: Missing spec

**Review by h4b00 — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/recipes_controller.rb [hunk 5]
```diff
@@ -39,7 +39,7 @@ def destroy
       private
 
       def recipe_params
-        params.require(:recipe).permit(%i[name content level cooking_time video_link price photo])
+        params.require(:recipe).permit([:name, :content, :level, :cooking_time, :video_link, :price, :photo, ingredients_attributes: [:quantity, :product_id]])
```
Comment (PatrykTyszka) [inline app/controllers/api/v1/recipes_controller.rb:42]: IMO you don't need to put it as an array.

File: app/serializers/ingredient_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class IngredientSerializer < ActiveModel::Serializer
```
Comment (PatrykTyszka) [inline app/serializers/ingredient_serializer.rb:1]: This file has wrong indentation

**Review by PatrykTyszka — APPROVED** [review]

File: app/serializers/recipe_serializer.rb (hunk 4 above)
Comment (plokijro) [reply app/serializers/recipe_serializer.rb]: I decided to create a serializer for ingredients and just add has_many :ingredients

File: db/migrate/20210822130020_create_ingredients.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CreateIngredients < ActiveRecord::Migration[6.1]
+  def change
+    create_table :ingredients do |t|
+      t.integer :quantity, null: false
```
Comment (h4b00) [inline db/migrate/20210822130020_create_ingredients.rb:4]:
```suggestion
      t.integer :quantity, null: false, default: 0
```

**Review by h4b00 — APPROVED** [review]

**Pull request merged**